### PR TITLE
Allow `.to()` to operate on tensors

### PR DESCRIPTION
**Pull request by Balandat**
Addresses #44.

### CODE REVIEW

File: linear_operator/utils/generic.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        device_args.add(kwargs["device"])
+
+    # Handle the case when casting is ambiguous
+    base_error_message = "Attempted to cast LinearOperator object to multiple"
+    if len(dtype_args) > 1:
+        raise RuntimeError(f"{base_error_message} dtypes ({dtype_args}.")
+    if len(device_args) > 1:
+        raise RuntimeError(f"{base_error_message} devices ({device_args}.")
+
+    dtype = next(iter(dtype_args)) if dtype_args else None
```
Comment (SebastianAment) [inline linear_operator/utils/generic.py]:
```suggestion
    dtype = dtype_args.pop() if dtype_args else None
```

File: linear_operator/utils/generic.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    device = next(iter(device_args)) if device_args else None
```
Comment (SebastianAment) [inline linear_operator/utils/generic.py]:
```suggestion
    device = device_args.pop() if device_args else None
```

File: test/utils/test_generic.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env python3
+
+import unittest
+
+import torch
+
+from linear_operator.test.base_test_case import BaseTestCase
+from linear_operator.utils.generic import _to_helper
+
+
```
Comment (SebastianAment) [inline test/utils/test_generic.py:10]: In order to test that this does the right thing for non-cpu tensors, could add a mock tensor like this:
```suggestion
class TestTensor(Tensor):
    @property
    def device(self):
        return self._device
```
Then in the test do 
```python
A = TestTensor()
A._device = torch.device("cuda")
```
`A` will then pretend to be on a CUDA device, without us having to have access to one.

**Review by SebastianAment — APPROVED** [review]
LGTM! Left comments inline. Could be worth adding the non-cpu tensor test before merging.

File: test/utils/test_generic.py (hunk 3 above)
Comment (Balandat) [reply test/utils/test_generic.py:10]: Yeah makes sense, or I can just use `mock` directly to assign the attribute.

**Pull request merged**